cli/cmd/credentials.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2021 SUSE
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"text/tabwriter"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+
+	client "github.com/SUSE/metabroker/apis/generated/clientset/versioned"
+)
+
+// NewDefaultCredentialsCmd creates a new credentials command with the default dependencies.
+func NewDefaultCredentialsCmd() *cobra.Command {
+	return NewCredentialsCmd(os.Stdout)
+}
+
+// NewCredentialsCmd creates a new credentials command.
+func NewCredentialsCmd(
+	stdout io.Writer,
+) *cobra.Command {
+	var namespace string
+	var timeout string
+	var noHeaders bool
+
+	cmd := &cobra.Command{
+		Use:  "credentials",
+		Args: cobra.ExactArgs(0),
+		// TODO: a better description.
+		Short: "Lists the Credentials on a given namespace.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			kubeconfigPath := viper.GetString("kubeconfig")
+
+			config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+			if err != nil {
+				return fmt.Errorf("failed to get credentials: %w", err)
+			}
+			clientset, err := client.NewForConfig(config)
+			if err != nil {
+				return fmt.Errorf("failed to get credentials: %w", err)
+			}
+
+			timeoutDuration, err := time.ParseDuration(timeout)
+			if err != nil {
+				return fmt.Errorf("failed to get credentials: %w", err)
+			}
+
+			ctx, cancel := context.WithTimeout(context.Background(), timeoutDuration)
+			defer cancel()
+
+			credentials, err := clientset.
+				ServicebrokerV1alpha1().
+				Credentials(namespace).
+				List(ctx, metav1.ListOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to get credentials: %w", err)
+			}
+
+			if len(credentials.Items) > 0 {
```

Comment:
Future feature request: `--output=json`.

Reply:
Added as an issue instead of TODO: https://github.com/SUSE/metabroker/issues/28.